api.tf

```
@@ -158,104 +213,84 @@ module "apim" {
   tags                      = var.tags
 }
 
-resource "azurerm_key_vault_certificate" "apim_proxy_endpoint_cert" {
-  name         = local.apim_cert_name_proxy_endpoint
-  key_vault_id = azurerm_key_vault.key_vault.id
-
-  certificate_policy {
-    issuer_parameters {
-      name = "Self"
-    }
+resource "azurerm_api_management_custom_domain" "api_custom_domain" {
+  api_management_id = module.apim.id
 
-    key_properties {
-      exportable = true
-      key_size   = 2048
-      key_type   = "RSA"
-      reuse_key  = true
-    }
+  proxy {
+    host_name    = trim(azurerm_private_dns_a_record.private_dns_a_record_api.fqdn, ".")
+    key_vault_id = azurerm_key_vault_certificate.apim_proxy_endpoint_cert.secret_id
```

```suggestion
    key_vault_id = azurerm_key_vault_certificate.apim_proxy_endpoint_cert.secret_id
    key_vault_id = replace(azurerm_key_vault_certificate.apim_proxy_endpoint_cert.secret_id, "/${azurerm_key_vault_certificate.apim_proxy_endpoint_cert.version}", "")
```
ref.
https://github.com/pagopa/io-infrastructure-modules-new/blob/master/azurerm_api_management/main.tf#L70

---

@pasqualedevita If I got it,  in IO infrastructure the certificate of the custom domain is set via a `hostname_configuration` argument block. In this case, instead, is set via the module `azurerm_api_management_custom_domain` which accepts only secret urls. If I try to apply the fix you suggested TF gives the following error `ApiManagement service only accepts KeyVault Secrets type`. I guess it's due to the fact that `replace` returns a (modified) certificate id not a secret key

---

the replace need remove pinned version certificate in keyvault url because when certificate will be renewed it creates a new version while apim point to older version.
Let's take me a look

---

the problem is: in azure portal on apim I see the pinned version and probably we'll need to do an apply to refresh the new version when certificate will be renewed into keyvault.

Can we use `azurerm_api_management.hostname_configurations` instead of `azurerm_api_management_custom_domain`?
@uolter what do u think?

---

Can we set the attribute reuse_key ? https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_certificate#reuse_key
It might not change the version.
Or we might set validity_in_months to 10 years. It's a self signed certificate used internally. It's not associated to a public endpoint. 
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_certificate#validity_in_months